server/bin/gen-certs

```diff
@@ -47,19 +81,153 @@ if [ ! -d $UPSTREAM_CERTS_HOME ]; then
     $SUDO mkdir -p $UPSTREAM_CERTS_HOME
 fi
 
-HOSTNAME=${HOSTNAME:-$(hostname)}
 
 if [ -f $CA_KEY ] && [ -f $CA_CERT ] && [ "$FORCECERT" != "1" ]; then
     echo "Certificates are already present."
 else
+    echo
+    echo "Creating CA private key password"
+    sudo su -c "echo $RANDOM > $CA_KEY_PASSWORD"
+
+    echo
     echo "Creating CA private key"
-    $SUDO openssl genrsa -out $CA_KEY 1024
+    sudo openssl genrsa -out $CA_KEY -passout "file:$CA_KEY_PASSWORD" 1024
+
+    echo
     echo "Creating CA public key"
-    $SUDO openssl rsa -pubout -in $CA_KEY -out $CA_PUB_KEY
+    sudo openssl rsa -pubout -in $CA_KEY -out $CA_PUB_KEY
+
+    echo
+    echo "Creating CA csr"
+    # this openssl.cnf is
+    sudo openssl req -config $OPENSSL_CONF -new -days 365 -key $CA_KEY -out $CA_CERT_CSR \
+       -subj "$CANDLEPIN_CA_SUBJ" \
+       -extensions v3_ca
+
+    echo
     echo "Creating CA certificate"
-    $SUDO openssl req -new -x509 -days $CA_CERT_DAYS -key $CA_KEY -out $CA_CERT -subj "/CN=$HOSTNAME/C=US/L=Raleigh/"
-    $SUDO su -c "echo -n "password" > $KEYSTORE_PASSWORD"
-    $SUDO openssl pkcs12 -export -in $CA_CERT -inkey $CA_KEY -out $KEYSTORE -name tomcat -CAfile $CA_CERT -caname root -chain -password file:$KEYSTORE_PASSWORD
-    $SUDO cp $CA_REDHAT_CERT $CA_UPSTREAM_CERT
-    $SUDO chmod a+r $KEYSTORE
+#    sudo openssl req -new -x509 -days 365 -key $CA_KEY -out $CA_CERT -subj \
+#        "/O=candlepinproject.org/OU=Candlepin CA/"
+    # this openssl.cnf is tweak to let us use lets DN info for our CA cert and signing requests
+    sudo openssl x509 -extfile $OPENSSL_CONF -extensions v3_ca \
+        -trustout -signkey $CA_KEY -days 365 -req -in $CA_CERT_CSR -out $CA_CERT
+
+    echo
+    echo "Creating apache server certs"
+    echo "Creating apache server private key password"
+    sudo su -c "echo $RANDOM > $SERVER_KEY_PASSWORD"
+
+    echo
+    echo "Creating apache server cert private key"
+    sudo openssl genrsa -out $SERVER_KEY -passout "file:$SERVER_KEY_PASSWORD" 1024
```

Comment:
Yeah, should have paramaterized that.

That code also only supports RSA priv keys and not EC keys. For the purposes of testing, suppose it wouldn't hurt to ge several (or at least a big RSA and a EC key for at least the server keys) and create corresponding certs, likely defaulting to the RSA based keys and certs.